File: Darts/src/code/bean/SpinnerSingleSelector.java
package code.bean;

import java.awt.Dimension;

import javax.swing.JSpinner;
import javax.swing.SpinnerNumberModel;
import javax.swing.event.ChangeEvent;
import javax.swing.event.ChangeListener;

public class SpinnerSingleSelector extends JSpinner
								   implements ChangeListener
{
	public SpinnerSingleSelector()
	{
		setSize(50, 25);
		setPreferredSize(new Dimension(50, 25));
		setModel(new SpinnerNumberModel(20, 1, 25, 1));
		addChangeListener(this);
	}
	
	/**
	 * This looks a bit weird, but what we want is:
	 * 
	 *  - Upping from 20 -> 25
	 *  - Downing from 25 -> 20
	 *  - Should not be able to enter 21-24 manually.
	 */
	@Override
	public void stateChanged(ChangeEvent arg0)
	{
		int val = (int)getValue();
		if (val == 21
		  || val == 22)
		{
			setValue(25);
		}
		else if (val > 22
		  && val < 25)
		{
			setValue(20);
		}
	}
}

File: src/main/kotlin/dartzee/bean/SpinnerSingleSelector.kt
package dartzee.bean

import java.awt.Dimension
import javax.swing.JSpinner
import javax.swing.SpinnerNumberModel
import javax.swing.event.ChangeEvent
import javax.swing.event.ChangeListener

class SpinnerSingleSelector : JSpinner(), ChangeListener {
    init {
        setSize(50, 25)
        preferredSize = Dimension(50, 25)
        model = SpinnerNumberModel(20, 1, 25, 1)
        addChangeListener(this)
    }

    /**
     * This looks a bit weird, but what we want is:
     * - Upping from 20 -> 25
     * - Downing from 25 -> 20
     * - Should not be able to enter 21-24 manually.
     */
    override fun stateChanged(arg0: ChangeEvent) {
        val intVal = value as Int
        when (intVal) {
            21,
            22 -> value = 25
            23,
            24 -> value = 20
        }
    }
}
